Enabling "graceful fail" for CellRanger `run_count` tasks (#357)
getwilds/wilds-wdl-library: pipelines/ww-sra-cellranger/README.md

| `create_bam` | Whether Cell Ranger should generate a BAM file | Boolean | No | true |
| `expect_cells` | Expected number of recovered cells per sample | Int | No | - |
| `chemistry` | Assay configuration (e.g., SC3Pv2, SC3Pv3) | String | No | auto-detect |

> | `skip_on_chemistry_failure` | If true, samples Cell Ranger can't detect a chemistry for are skipped instead of failing the workflow. See [Mixed single-cell / non-single-cell input](#mixed-single-cell--non-single-cell-input). | Boolean | No | `false` |

Reviewer: ```suggestion
| `skip_on_chemistry_failure` | If true, samples Cell Ranger can't auto-detect a chemistry for are skipped instead of failing the workflow. See the module [README](../../modules/ww-cellranger/README.md#graceful-chemistry-detection-skip). | Boolean | No | `false` |
```
should specify the auto detection I think. Also, may as well link to the primary description of this flag instead of describing it in two places slightly differently.

| Output | Description | Source Module |
|--------|-------------|---------------|
| `single_cell_sample_list` | Newline-delimited list of sample IDs that ran Cell Ranger successfully | pipeline |
| `skipped_sample_list` | Newline-delimited list of sample IDs skipped due to chemistry-detection failure (empty unless `skip_on_chemistry_failure = true`) | pipeline |
| `cellranger_results` | Compressed tarballs of Cell Ranger count output directories (successful samples only) | ww-cellranger |
| `cellranger_web_summaries` | Web summary HTML files (successful samples only) | ww-cellranger |
| `cellranger_metrics` | Metrics summary CSV files (successful samples only) | ww-cellranger |

> | `cellranger_filtered_h5s` | Filtered feature-barcode matrix HDF5 files (successful samples only) | ww-cellranger |

Reviewer: ```suggestion
| `single_cell_sample_list` | List of sample IDs that Cell Ranger ran successfully | pipeline |
| `skipped_sample_list` | List of sample IDs that Cell Ranger failed to auto-detect chemistry for (empty unless `skip_on_chemistry_failure = true`) | pipeline |
| `cellranger_results` | Compressed tarballs of Cell Ranger count output directories | ww-cellranger |
| `cellranger_web_summaries` | Web summary HTML files | ww-cellranger |
| `cellranger_metrics` | Metrics summary CSV files | ww-cellranger |
| `cellranger_filtered_h5s` | Filtered feature-barcode matrix HDF5 files | ww-cellranger |
```
I think "list" implies newline-delimited (and uses less jargon). I also think it's implied that outputs are generated if the task is successful, and we state that if chemistry detection fails then the 'usual' outputs are not produced.

Setting `skip_on_chemistry_failure = true` (default `false`) makes such samples succeed with no count outputs instead. Their IDs are written to `skipped_sample_list`, the IDs of samples that completed normally go to `single_cell_sample_list`, and the `cellranger_*` output arrays contain results only for the successful samples. Other Cell Ranger failures (OOM, corrupt reference, etc.) always re-raise regardless of this flag — see [the module README](../../modules/ww-cellranger/README.md#graceful-chemistry-detection-skip) for detection details.

> **Heads up — `max_reads` interaction:** If you set `max_reads` low enough that the downsampled FASTQs fall below Cell Ranger's per-chemistry minimum read-length requirements, even a legitimately single-cell sample will trip the skip heuristic and land in `skipped_sample_list`. This is generally only an issue at very small `max_reads` values (a few hundred thousand or less). If you're testing with small inputs and seeing real single-cell samples skipped, raise `max_reads`.

> > **Important:** `skip_on_chemistry_failure` only works when `chemistry` is left unset. Specifying `chemistry` (e.g. `"SC3Pv3"`) tells Cell Ranger to skip auto-detection, so a non-single-cell sample will fail with a different error that the graceful-skip heuristic does not catch. If you have a mixed scatter, leave `chemistry` unset so Cell Ranger auto-detects per sample.

Reviewer: ```suggestion
```
I think it's better to describe this in one place (the module README) rather than two. Anything else you want to say about it should be added to that description instead of covered separately here in the pipeline.

Author: Agreed, some of this section is duplication from the ww-cellranger module, but the first half feels relevant and specific to the workflow. Updated in https://github.com/getwilds/wilds-wdl-library/pull/357/commits/7c073d2e83baf4aada038eede94023d9c919deb0 to be much more concise.
